File: LeetCode/234. Palindrome Linked List/elixir/solution.exs
# Definition for singly-linked list.
#
# defmodule ListNode do
#   @type t :: %__MODULE__{
#           val: integer,
#           next: ListNode.t() | nil
#         }
#   defstruct val: 0, next: nil
# end

defmodule Solution do
  @spec is_palindrome(head :: ListNode.t() | nil) :: boolean
  def is_palindrome(head) do
    head
    |> to_list()
    |> then(&(&1 == Enum.reverse(&1)))
  end

  defp to_list(node, acc \\ [])
  defp to_list(nil, acc), do: acc
  defp to_list(node, acc), do: to_list(node.next, [node.val | acc])
end

File: LeetCode/234. Palindrome Linked List/java/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        List<Integer> list = new ArrayList<>();

        ListNode node = head;
        while (node != null) {
            list.add(node.val);
            node = node.next;
        }

        List<Integer> reversedList = new ArrayList<>(list);
        Collections.reverse(reversedList);

        return list.equals(reversedList);
    }
}
